torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1977,91 +2058,34 @@ void ProcessGroupNCCL::watchdogHandler() {
       if (!terminateProcessGroup_.load()) {
         work.checkAndSetException();
       }
+      // Then check if work has timed out
+      bool timedout = work.checkTimeout();
```

Comment:
This basically changes the existing error logic which intentionally differentiate nccl errors from timeout. There is benefit of doing so that we can identify the type of error immediately from the LOGs.

Reply:
In `checkTimeout`, exception will not be changed if it has been set.

Reply:
Understood. Not just the exception logging, I was talking about the logging within the `if (work.exception()) {} `branch. Previously, it would be logged only for nccl errors, and only if there is no nccl errors, we checked the timeout, and timeout error would have a different logging. It's very handy in debugging when we only log the first/true errors.  If we found an nccl error, there is no need to check the timeout. that's why I think the existing if/else structure is preferred.

Reply:
Got it. I believe `checkAndSetException` has logging too?

Reply:
Yeah, I checked, it has logging.